NitroxModel/Extensions.cs

```diff
@@ -8,13 +11,39 @@ public static class Extensions
         public static TAttribute GetAttribute<TAttribute>(this Enum value)
             where TAttribute : Attribute
         {
-            var type = value.GetType();
-            var name = Enum.GetName(type, value);
+            Type type = value.GetType();
+            string name = Enum.GetName(type, value);
 
-            return type.GetField(name) 
+            return type.GetField(name)
                 .GetCustomAttributes(false)
                 .OfType<TAttribute>()
                 .SingleOrDefault();
         }
+
+        public static bool HasStateFlag(this PlayerSlotReservationState currentState, PlayerSlotReservationState checkedState)
```

Comment:
How about moving these `PlayerSlotReservationState`-specific functions to the same file as the enum itself? I see this class as a container for general extension methods that can be used on multiple places throughout the program, not specifically for a single enum. Thoughts?